### PR TITLE
Conversation lifecycle improvements

### PR DESCRIPTION
**Pull request by jerith**
We currently have routing and certain conversation operations (such as bulk send) tied to conversation lifecycle transitions. We need to decouple these in the backend to make the new routing and UI features possible.

### CODE REVIEW

File: go/apps/bulk_message/vumi_app.py [hunk 1]
```diff
@@ -66,20 +66,23 @@ def send_message_via_window(self, conv, window_id, batch_id, to_addr,
             'msg_options': msg_options,
             })
 
+    def process_command_initial_action_hack(self, *args, **kwargs):
+        return self.process_command_bulk_send(*args, **kwargs)
```
Comment (hodgestar) [inline go/apps/bulk_message/vumi_app.py]: Can we add a TODO here to remove this once the conversation start/end stuff that calls it is gone?

File: go/apps/surveys/views.py [hunk 2]
```diff
@@ -133,8 +133,8 @@ def contents(request, conversation_key):
 
     survey_form = make_read_only_form(ConversationForm(request.user_api,
         instance=conversation, initial={
-            'start_date': conversation.start_timestamp.date(),
-            'start_time': conversation.start_timestamp.time(),
+            'start_date': conversation.created_at.date(),
+            'start_time': conversation.created_at.time(),
```
Comment (hodgestar) [inline go/apps/surveys/views.py:125]: I think we should remove `start_date` and `start_time` entirely even though this means editing the form and templates a bit.

File: go/apps/surveys/vumi_app.py [hunk 3]
```diff
@@ -130,33 +130,36 @@ def end_session(self, participant, poll, message):
...
-            log.error("No account key in batch metadata: %r" % (batch,))
-            return
+    #     user_account_key = batch.metadata["user_account"]
+    #     if user_account_key is None:
+    #         log.error("No account key in batch metadata: %r" % (batch,))
+    #         return
+
+    #     user_api = self.get_user_api(user_account_key)
+    #     conv = yield user_api.get_wrapped_conversation(conversation_key)
+    #     returnValue(conv)
```
Comment (hodgestar) [inline go/apps/surveys/vumi_app.py]: Can we just remove this commented out code?

**Comment by hodgestar** [comment]
:+1:

**Comment by smn** [comment]
:+1:

**Pull request merged**